Adds a document about the fakes extension in MSTest Runner (#41517)
dotnet/docs: docs/core/testing/unit-testing-platform-extensions-fakes.md

---
title: Microsoft.Testing.Platform Fakes extension

> description: Learn about the various Microsoft.Testing.Platform Fakes extension and how to use it.

Reviewer: ```suggestion
description: Learn about the various Microsoft.Testing.Platform Fakes extension capabilities and how to use it.
```

author: drognanar
ms.author: arturs
ms.date: 06/21/2024
---

# Fakes extension

> This extension provides support to execute a test project that makes use of `Microsoft Fakes`.

Reviewer: ```suggestion
The `Microsoft.Testing.Extensions.Fakes` extension provides support to execute a test project that makes use of `Microsoft Fakes`.
```

> This extension provides suppoort to execute a test project that makes use of `Microsoft Fakes`.

Reviewer 2: ```suggestion
This extension provides support to execute a test project that makes use of `Microsoft Fakes`.
```

Author: fixed

> [Microsoft Fakes](https://learn.microsoft.com/en-us/visualstudio/test/isolating-code-under-test-with-microsoft-fakes) allows you to better test your code by either generating `Stub`s (for instance creating a testable implementation of `INotifyPropertyChanged`) or by `Shim`ing methods and static methods (replacing the implementation of `File.Open` with a one you can control in your tests).

Reviewer 2: You should be using relative links to the md file.

Author: the thing is that I could not find the fakes documentation in this repository.

there's also the `aka.ms/msfakes` shortlink that could be used instead.

> [!NOTE]
> This extension requires a VS Enterprise installation with the minimum version of 17.11 preview 1 in order to work correctly.

> # Upgrading your project to the new extension

Reviewer 2: Only one header should be using top level header

Author: fixed
